packages/slate/src/models/node.js

```diff
@@ -943,25 +957,29 @@ class Node {
    * @return {Array<String>}
    */
 
-  getKeysAsArray() {
-    const keys = []
+  getKeysAsDictionary() {
+    const keys = {}
 
-    this.forEachDescendant(desc => {
-      keys.push(desc.key)
-    })
+    // PERF: prevent JS to search at prototype chain, performance concern
+    keys.__proto__ = null
+    keys[this.key] = true
 
-    return keys
-  }
+    this.nodes.forEach(child => {
+      if (child.object === 'text') {
+        keys[child.key] = true
+        return
+      }
 
-  /**
-   * Return a set of all keys in the node.
-   *
-   * @return {Set<String>}
-   */
+      // PERF: use the logic of utils._extend; becuase
```

Comment:
Can you remove this and add it back in a later pull request to prove that it makes a big enough difference to merit adding?

Reply:
Fixed~